### PR TITLE
Bug 1311147 - Implement a default blacklist of domains to not appear in highlights

### PR DESCRIPTION
**Pull request by bkmunar**

### CODE REVIEW

**Comment by farhanpatel** [comment]
Don't forget the tests 😏  There is already a `TestSQLiteHistoryRecommendations.swift` where you can add tests to.

**Comment by farhanpatel** [comment]
I think a good starting for a blacklist would be using the installed SearchEngines. You'll also get the user's local variation of Google as well :)

File: Storage/SQL/SQLiteHistoryRecommendations.swift [hunk 1]
```diff
@@ -18,6 +18,37 @@ extension SQLiteHistory: HistoryRecommendations {
...
+            "calendar.google.com",
+            "mail.google.com",
+            "mail.yahoo.com",
+            "search.yahoo.com",
+            "localhost",
+            "t.co"
+        ]
+
+        var blacklistSubquery = ""
+        if (blacklistedHosts.count > 0) {
```
Comment (sleroux) [inline Storage/SQL/SQLiteHistoryRecommendations.swift]: Nit: `BrowserDB` has a class method named `varlist` that creates the (?,?,?..?) string you need so you don't need to do this here.

File: Storage/SQL/SQLiteHistoryRecommendations.swift [hunk 2]
```diff
@@ -18,6 +18,37 @@ extension SQLiteHistory: HistoryRecommendations {
         let thirtyMinutesAgo = NSNumber(unsignedLongLong: now - 30 * microsecondsPerMinute)
         let threeDaysAgo = NSNumber(unsignedLongLong: now - (60 * microsecondsPerMinute) * 24 * 3)
 
+        let blacklistedHosts: [AnyObject?] = [
```
Comment (farhanpatel) [inline Storage/SQL/SQLiteHistoryRecommendations.swift]: You can probably remove the AnyObject here.

**Comment by bkmunar** [comment]
up for review again when you get back @sleroux

File: StorageTests/TestSQLiteHistoryRecommendations.swift [hunk 3]
```diff
@@ -134,6 +134,61 @@ class TestSQLiteHistoryRecommendations: XCTestCase {
...
+
+        /*
+         * Site A: 1 visit, 1 hour ago = highlight
+         * Site B: 1 visits, 15 minutes ago = non-highlight
+         * Site C: 3 visits, 1 hour ago = highlight
+         * Site D: 4 visits, 1 hour ago = non-highlight
+         */
+        let siteA = Site(url: "http://www.google.com", title: "A")
+        let siteB = Site(url: "http://siteB/", title: "B")
+        let siteC = Site(url: "http://www.search.yahoo.com/", title: "C")
```
Comment (sleroux) [inline StorageTests/TestSQLiteHistoryRecommendations.swift]: Nit: I would add a comment clarifying that yahoo/google domains are on the blacklist. Without that, it looks like this test doesn't actually do much.

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
ButtonToast.swift | 18 | Colons should be next to the identifier when specifying a type. 
#### Errors

File | Line | Reason |
| --- | ----- | ----- |
ButtonToast.swift | 23 | Opening braces should be preceded by a single space and on the same line as the declaration. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Pull request merged**